pkg/apis/extensions/v1alpha1/types_controlplane.go

```diff
@@ -67,6 +67,8 @@ type ControlPlaneSpec struct {
 	DefaultSpec `json:",inline"`
 	// Purpose contains the data if a cloud provider needs additional components in order to expose the control plane.
 	// This field is immutable.
+	// Deprecated: This field will be removed in a future release.
```

Comment:
I usually add an empty line between the doc string and the deprecation message to visually distinguish them.

```suggestion
	// This field is immutable.
	//
	// Deprecated: This field will be removed in Gardener v1.123.
```

However, it is subjection and I am fine if you prefer to keep it like it is.